portfolio/src/main/webapp/CreatePuzzlePage.js

```diff
@@ -4,18 +4,26 @@ const responseMessage = document.getElementById("upload-message");
 //Handles the Create Puzzle Form Submission
 const handleCreatePuzzleFormSubmission = (e) =>{
     e.preventDefault();
-    createPuzzleForm.reset();
 
     //Getting the link to the newly created puzzle
     getPuzzleLink();
+
+    createPuzzleForm.reset();
 }
 
 async function getPuzzleLink () {
-    try{
-        const response = await fetch("/upload");
+    const data = new FormData(createPuzzleForm);
+    const difficulty = data.get("difficulty");
+    data.set("difficulty", difficulty.toUpperCase());
```

Comment:
This definitely works. Another approach would be to apply the `value` attribute to the option elements in createpuzzle.html:

```html
<select class="form-control" id="puzzle-difficulty" name="difficulty">
        <option value="EASY">Easy</option>
        <option value="MEDIUM">Medium</option>
        <option value="HARD">Hard</option>
</select>
```
This has the benefit of decoupling the actual values from how they are presented to the users. For example we could change "Medium" to "Moderate" without having to change the value of the enum on the backend.